z3/src/z3_term.cpp

```diff
@@ -104,26 +104,159 @@ bool Z3Term::compare(const Term & absterm) const
+        case Z3_OP_FPA_ABS: return Op(Abs);
+        case Z3_OP_FPA_TO_REAL: return Op(To_Real);
+        case Z3_OP_STR_TO_INT: return Op(To_Int);
+        case Z3_OP_IS_INT: return Op(Is_Int);
+        case Z3_OP_BNOT: return Op(BVNot);
+        case Z3_OP_BNEG: return Op(BVNeg);
+        // binary
+        case Z3_OP_XOR: return Op(Xor);
+        case Z3_OP_IMPLIES: return Op(Implies);
+        case Z3_OP_POWER: return Op(Pow);
+        case Z3_OP_IDIV: return Op(IntDiv);
+        case Z3_OP_DIV: return Op(Div);
+        case Z3_OP_LT: return Op(Lt);
+        case Z3_OP_FPA_ROUND_TO_INTEGRAL: return Op(To_Int);
+        case Z3_OP_LE: return Op(Le);
+        case Z3_OP_GT: return Op(Gt);
+        case Z3_OP_GE: return Op(Ge);
+        case Z3_OP_EQ: return Op(Equal);
+        case Z3_OP_MOD: return Op(Mod);
+        case Z3_OP_CONCAT: return Op(Concat);
+        case Z3_OP_BAND: return Op(BVAnd);
+        case Z3_OP_BOR: return Op(BVOr);
+        case Z3_OP_BXOR: return Op(BVXor);
+        case Z3_OP_BNAND: return Op(BVNand);
+        case Z3_OP_BNOR: return Op(BVNor);
+        case Z3_OP_BXNOR: return Op(BVXnor);
+        case Z3_OP_BADD: return Op(BVAdd);
+        case Z3_OP_BSUB: return Op(BVSub);
+        case Z3_OP_BMUL: return Op(BVMul);
+        case Z3_OP_BUDIV: return Op(BVUdiv);
+        case Z3_OP_BUREM: return Op(BVUrem);
+        case Z3_OP_BSDIV: return Op(BVSdiv);
+        case Z3_OP_BSREM: return Op(BVSrem);
+        case Z3_OP_BSMOD: return Op(BVSmod);
+        case Z3_OP_BSHL: return Op(BVShl);
+        case Z3_OP_BASHR: return Op(BVAshr);
+        case Z3_OP_BLSHR: return Op(BVLshr);
+        case Z3_OP_ULT: return Op(BVUlt);
+        case Z3_OP_ULEQ: return Op(BVUle);
+        case Z3_OP_UGT: return Op(BVUgt);
+        case Z3_OP_UGEQ: return Op(BVUge);
+        case Z3_OP_SLEQ: return Op(BVSle);
+        case Z3_OP_SLT: return Op(BVSlt);
+        case Z3_OP_SGEQ: return Op(BVSge);
+        case Z3_OP_SGT: return Op(BVSgt);
+        case Z3_OP_ROTATE_LEFT: return Op(Rotate_Left);
+        case Z3_OP_ROTATE_RIGHT: return Op(Rotate_Right);
+        case Z3_OP_SELECT: return Op(Select);
+        // ternary
+        case Z3_OP_ITE: return Op(Ite);
+        case Z3_OP_STORE: return Op(Store);
+        // variadic
+        case Z3_OP_AND: return Op(And);
+        case Z3_OP_OR: return Op(Or);
+        case Z3_OP_ADD: return Op(Plus);
+        case Z3_OP_SUB: return Op(Minus);
+        case Z3_OP_MUL: return Op(Mult);
+        case Z3_OP_DISTINCT: return Op(Distinct);
+        // indexed
+        case Z3_OP_EXTRACT: return Op(Extract);
+        case Z3_OP_ZERO_EXT: return Op(Zero_Extend);
+        case Z3_OP_SIGN_EXT: return Op(Sign_Extend);
+        case Z3_OP_REPEAT: return Op(Repeat);
+        case Z3_OP_INT2BV: {
+          size_t out_width = range.bv_size();
+          return Op(Int_To_BV, out_width);
+        }
+        case Z3_OP_BV2INT:
+          return Op(BV_To_Nat);
+          // Op(Apply) not handled...
```

Comment:
If there's a way to check whether something is a function application, you could do that above. Doesn't have to be in this PR, just mentioning it.